### PR TITLE
Data model #39

### LINKED ISSUE #39: Implement approved data model
If data model from task/issue #29 is approved, it can be implemented in the backend.
- Change and create the needed scaffolding classes and code for creating new/extended tables
- Create ER-model out of mysql db for documentation

### PR DESCRIPTION
**Pull request by stfnltnr**
resolves #39 

The following tables, as defined within the data model design document, are created during start up of the GateWayService:

- UserTable
- OrganizationTable
- ProcessStoreTable
- Orga_Process_Map - mapping between Organizations and Processes
- RatingTable
- ReviewTable

At the moment all tables are created within the ippr_security schema since their is no separete StoreService or schema. Only the data model is created for now.

Tested only with login of existing Users from user.csv. Seems that the changes don't cause any trouble :see_no_evil:, of course there aren"t any crud operations on the new datamodel so far :construction: .

**Please review the created data model (Tables, FKs, ...) with mySql Workbench or something similar since their is no opportunity to test it elsewhere atm**

If it is ok we can proceed with the next steps:
- modify RBACService for login and registration via db instead of cached users (Issue #40)
- provide ShopService - discussion needed how we plan to do the exactly
- extend Services that is possible to provide Rest-Endpoints to perform CRUD operations on added tables

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ This file ONLY contains changes made by the AIM17 dev-team. This means, that thi
 of the forked repository.
 The format is partially based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
+## 2018-04-24 ([#39](https://github.com/amarbajric/EBUSA-AIM17/pull/39))
```
Comment (amarbajric) [inline CHANGELOG.md]: Please change the url `https://github.com/amarbajric/EBUSA-AIM17/pull/39` to `https://github.com/amarbajric/EBUSA-AIM17/pull/68` as you are referring to your Pull Request :)
Also change the `#39` to `#68`.

So it should look like this:
`## 2018-04-24 ([#68](https://github.com/amarbajric/EBUSA-AIM17/pull/68))`

**Review by amarbajric — CHANGES_REQUESTED** [review]
**REVIEW**
I have generated a new ER-Diagram of the `ippr_security` table by using the mysql workbench. It can be found on the SP under `Development/Data_model/ippr_security_new.pdf`
Also, I have reviewed the tables and relations and I have only found one minor point regarding the `review` table. Everything else looks very promising to me.
- `review` table does not contain attribute for storing the `reviewer/approver`. Is this on purpose or have you forgotten about this point?

I have also found a very minor mistake in the `CHANGELOG.md` file that I have commented. Other than that, I think we can merge this branch into the `dev` branch as soon as these two points have been fixed/discussed. Thank you @loete 👍

**Comment by stfnltnr** [comment]
Ok, missing columns was a naming mistake within `ReviewImpl.java` - column u_id is now replaced by to columns `uploader` and `approver`.
Changed link in `CHANGELOG.md` to PR #68 

Pls give it another try :crossed_fingers:

**Comment by amarbajric** [comment]
Thank you for the fast response and committed changes!
Now everything looks fine to me and can be merged!
**APPROVED** 👍

**Review by amarbajric — APPROVED** [review]

**Comment by tortmann** [comment]
> Thank you for the fast response and committed changes!
> Now everything looks fine to me and can be merged!
> APPROVED 👍

Also approved from my side since I didnt't find any additional "mistakes"

>If it is ok we can proceed with the next steps:
>- modify RBACService for login and registration via db instead of cached users (Issue #40)
>- provide ShopService - discussion needed how we plan to do the exactly
>- extend Services that is possible to provide Rest-Endpoints to perform CRUD operations on added tables

Next steps are also approved, where **modifying the RBACService (#40) has priority** so we can hopefully present a working login and register form with the new backend during the interim presentation on Friday, 27.04

ER-diagram of implemented data model: SharePoint **Documents\Development\Data_model\ippr_security_new.pdf**

**Pull request merged**